### PR TITLE
Insert many documents support

### PR DESCRIPTION
**Pull request by tomgro**
As Patryk mentioned in point 2 https://groups.google.com/forum/#!topic/vertx/fWeg5yExceA
there is no possibility to insert list of documents. 
I've added new methods and I've accepted Eclipse CLA ;)

### CODE REVIEW

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/InsertManyOptions.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Options for configuring inserts.
+ *
+ * @author <a href="mailto:tomasz.groch@gmail.com">Tomasz Groch</a>
+ */
+@DataObject
+public class InsertManyOptions {
```
Comment (purplefox) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/InsertManyOptions.java]: If this only contains one field "ordered" do we really need a whole new class for this? Can't a boolean param just be specified in the insertManyXXX methods?

Comment (tomgro) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/InsertManyOptions.java]: You are right, I will fix it.

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java [hunk 2]
```diff
@@ -107,6 +107,50 @@ static MongoClient createShared(Vertx vertx, JsonObject config) {
   MongoClient insertWithOptions(String collection, JsonObject document, WriteOption writeOption, Handler<AsyncResult<String>> resultHandler);
 
   /**
+   * Insert a documents in the specified collection
```
Comment (purplefox) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java]: Insert documents

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java [hunk 3]
```diff
@@ -107,6 +107,50 @@ static MongoClient createShared(Vertx vertx, JsonObject config) {
...
+  /**
+   * Insert a documents in the specified collection with the specified many options
+   *
+   * @param collection  the collection
+   * @param documents  the documents
+   * @param ordered  the insert many options field to use
+   * @param resultHandler  will be called when complete
+   */
+  @Fluent
+  MongoClient insertManyWithManyOptions(String collection, List<JsonObject> documents, boolean ordered, Handler<AsyncResult<Void>> resultHandler);
```
Comment (purplefox) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java]: ManyOptions has gone so method should be renamed

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java [hunk 4]
```diff
@@ -107,6 +107,50 @@ static MongoClient createShared(Vertx vertx, JsonObject config) {
...
+   * Insert a documents in the specified collection with the specified write many options and write option
+   *
+   * @param collection  the collection
+   * @param documents  the documents
+   * @param ordered  the insert many options field to use
+   * @param writeOption  the write option to use
+   * @param resultHandler  will be called when complete
+   */
+  @Fluent
+  MongoClient insertManyWithManyOptionsAndWriteOption(String collection, List<JsonObject> documents, boolean ordered, WriteOption writeOption, Handler<AsyncResult<Void>> resultHandler);
```
Comment (purplefox) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java]: rename

File: vertx-mongo-client/src/main/asciidoc/java/index.adoc [hunk 5]
```diff
@@ -167,6 +167,26 @@ mongoClient.insert("books", document, res -> {
 });
 ----
 
+To insert a documents you use `link:../../apidocs/io/vertx/ext/mongo/MongoClient.html#insertMany-java.lang.String-java.util.List-io.vertx.core.Handler-[insertMany]`.
```
Comment (purplefox) [inline vertx-mongo-client/src/main/asciidoc/java/index.adoc]: This file is auto-generated when you build the project from package-info.java.

You should add the documentation in package-info.java.

Also please make sure you build and test the project before you push.

mvn clean install

**Comment by karianna** [comment]
@tomgro Are you able to pull from master and resolve any merge conflicts and push this branch again?

**Comment by karianna** [comment]
@tomgro - Bump :-)

**Comment by karianna** [comment]
@tomgro I'll leave this open for a week and then close as out of date.

**Pull request closed without merging**